test/e2e/operator.go

```diff
@@ -272,3 +277,102 @@ var _ = Describe("ARO Operator - Conditions", func() {
 		Expect(err).NotTo(HaveOccurred())
 	})
 })
+
+var _ = Describe("ARO Operator - Azure NSG Reconciler", func() {
+	var location, resourceGroup, subnet, vnet string
+	var correctNSG *string
+	var dummynsg mgmtnetwork.SecurityGroup
+	ctx := context.Background()
+	type subnetDescriptor struct {
+		resourceGroup string
+		vnetName      string
+		subnetName    string
+		location      string
+	}
+	enableReconcileNSG := func() {
+		instance, err := clients.AROClusters.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		if !instance.Spec.Features.ReconcileNSGs {
+			instance.Spec.Features.ReconcileNSGs = true
+			_, err = clients.AROClusters.AroV1alpha1().Clusters().Update(ctx, instance, metav1.UpdateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		}
+	}
+	getDescriptorFromProviderSpec := func(providerSpec *runtime.RawExtension) (*string, *subnetDescriptor, error) {
+		var spec azureproviderv1beta1.AzureMachineProviderSpec
+		err := json.Unmarshal(providerSpec.Raw, &spec)
+		if err != nil {
+			return nil, nil, err
+		}
+		return &spec.ResourceGroup, &subnetDescriptor{
+			resourceGroup: spec.NetworkResourceGroup,
+			vnetName:      spec.Vnet,
+			subnetName:    spec.Subnet,
+			location:      spec.Location,
+		}, nil
+	}
+	gatherNetworkInfo := func() {
+		machines, err := clients.MachineAPI.MachineV1beta1().Machines("openshift-machine-api").List(ctx, metav1.ListOptions{LabelSelector: "machine.openshift.io/cluster-api-machine-role=worker"})
```

Comment:
This reconciler reconciles both worker and master subnets. Is there a benefit of testing both in e2e?

Reply:
Yes we should probably test both. I will need to modify the `dummyNSG` properties because when I changed it on the master subnet the tests failed because it could not connect to the cluster anymore.

Reply:
I separated into testing reconcile on worker subnet and master subnet